### ISSUE TITLE
[KVNProgress Show] -> [KVNProgress ShowSuccess] animation glitch?

### ISSUE THREAD

**Issue by JustinDMoore** [kevin-hirsch/KVNProgress#13]
Sorry to add more to your pile ;) I noticed that there is an animation issue going on when transitioning from [KVNProgress Show] to [KVNProgress ShowSuccess]. The circle doesn't "morph" into the success checkmark as it used to. It seems as though a new "success" view is created and animated into place.  

Here is a link to a recording of what I am experiencing: 
https://www.youtube.com/watch?v=bAFHN-BnSBI&feature=youtu.be

Here is the complete code:

   ```objc
    -(void)saveCoverPhoto:(UIImage *)photo {
        
        [KVNProgress show];
        
        NSData *imageData = UIImagePNGRepresentation(photo);
        PFFile *imageFile = [PFFile fileWithName:@"picture.png" data:imageData];
        [imageFile saveInBackgroundWithBlock:^(BOOL succeeded, NSError *error) {
            if (!error) {
                if (succeeded) {
                    PFUser *user = [PFUser currentUser];
                    if (coverPhoto) {
                        user[@"coverPicture"] = imageFile;
                    } else {
                        user[@"picture"] = imageFile;
                        user[@"thumbnail"] = imageFile;
                    }
                    
                    [user saveInBackgroundWithBlock:^(BOOL succeeded, NSError *error) {
                        if (succeeded) {
                         
                            [self initUI];
                            [KVNProgress showSuccess];
                        } 
                    }];
                }
            } else {
                [KVNProgress showErrorWithStatus:@"Could not update picture"];
            }
        }];
    }
   ```

**Comment by kevin-hirsch** [comment]
I have been away for a week. I am back now. I'm going to check this issue this week.

**Comment by JustinDMoore** [comment]
Were you able to duplicate this? Just following up on it. We're beginning beta in a week or two and like to resolve this.

**Comment by ZaneH** [comment]
It's because it's in a block right?

**Comment by JustinDMoore** [comment]
I don't see why calling success from a block makes a difference. Can you elaborate?

**Comment by ZaneH** [comment]
Blocks are run on a different thread and since the methods are class methods it's initializing a new object for the success method. I could be wrong, but I'm fairly sure that's what's happening.

**Comment by JustinDMoore** [comment]
Good catch. I'm going to play with it when I get home in a couple hours. Could explain another issue I'm having.

**Comment by kevin-hirsch** [comment]
Hello @JustinDMoore, calling:

```objc
[KVNProgress show];
[KVNProgress showSuccess];
```

is causing me the same glitch as you have. 
I don't that calling it from a block is what causes a problem but @ZaneH is right, blocks are running on different threads (usually not the main one) and all UI is supposed to be executed on the main thread, if not, it can causes UI bugs or even do not make run your UI changes at all. So, when in a block, don't forget to put your UI code in a:

```objc
dispatch_async(dispatch_get_main_queue(), ^{
    // Code that will execute on the main thread
    // Put your UI here
});
```

Putting that aside, there's still a problem: the glitch. I will try to see from where it is coming.

I also saw in your video that you don't have any background color to the loading circle, is that on purpose or is it a bug?

**Comment by kevin-hirsch** [comment]
@JustinDMoore, can you download latest version of the repo (or pod 2.2.1) and test if everything is fixed for you? Thanks! :)

**Comment by JustinDMoore** [comment]
@kevin-hirsch I did wrap my UI related in the main thread last night. I should have known better ;) Only after updating to 2.2.1 was the problem resolved. Thanks for your work on this. It's really good.

**Comment by kevin-hirsch** [comment]
Thank you for your feedback and for finding the bug ! :) :+1:
